### ISSUE TITLE
[BUG] retrieve BU folder with label-less entry?

### ISSUE THREAD

**Issue by JoernBerkefeld** [Accenture/sfmc-devtools-vscode#169]
![image](https://github.com/Accenture/sfmc-devtools-vscode/assets/1917227/49ff0ef1-9be8-4049-9bf4-4e10119ccc08)

after i right clicked a BU folder and selected mcdev-retrieve

2 issues really:
- one entry after SenderProfile without label
- sorting of entries weird. please apply alphabetical sorting

**Comment by JoernBerkefeld** [comment]
good thinking. there is actually an empty element in my dev-version...
```
{
    "name": "",
    "apiName": "",
    "retrieveByDefault": true,
    "supports": {
      "retrieve": true,
      "create": true,
      "update": true,
      "delete": true,
      "changeKey": false,
      "buildTemplate": true,
      "retrieveAsTemplate": false
    },
    "description": "here as a placeholder to have auto completion"
  },
```

I shall filter that in mcdev.

with that said - your list is somehow out of order - the returned json array is actually already sorted alphabetically but your list vscode is not?

**Comment by anasilva105** [comment]
@JoernBerkefeld I checked and it seems that when running ``mcdev et`` the array is sorted by the apiName and not the name of the metadata type. This is why it's unsorted since I'm showing the name.

I will sort it in the extension.

**Comment by JoernBerkefeld** [comment]
tracking the underlying problem in https://github.com/Accenture/sfmc-devtools/issues/1306

**Comment by JoernBerkefeld** [comment]
issue will be resolved with 6.1 release:
- https://github.com/Accenture/sfmc-devtools/pull/1309
